### ISSUE TITLE
When Backup target is delete , we are allowed to use backups existing in this target.

### ISSUE THREAD

**Issue by sangeethah** [rancher/longhorn#159]
Tested with - longhorn version - cjellick/longhorn:dev-740fa39  and latest cattle build form jimeng's branch.

Currently deleting a backup target is just marking the state as "removed" in DB (backups are not deleted from the backup store), so it is not available for creating more backups . 
We are still allowed to use the existing backups and it continues to work as long as the data is available in the mount points. 

Would the right thing to do be to  mark the backups that exist in the removed backup target as "removed" as well , so we do not permit any action on them?

**Comment by cjellick** [comment]
Going to ask for more input from @ibuildthecloud and @vincent99 on this.

But I don't think auto-deleting the backups is the right thing to do.

**Comment by vincent99** [comment]
Yes, definitely shouldn't destroy the data.  You could maybe hide the restore action if the backupTarget of the backup is removed?

**Comment by cjellick** [comment]
@vincent99 I guess my thought was that you should still be able to use a backup even if the target was removed, you just cant use that target for new backups.

I don't know if that's right, but if we make restore unavailable, the backup is useless.

**Comment by vincent99** [comment]
But if the target is removed you have no reason to think that it's still a valid place to retrieve that backup from...

Disallow remove on targets that have any un-removed backups instead?
